### PR TITLE
Use SimpleTables get_stat_data for Wazi ECD

### PR DESCRIPTION
**Pull request by longhotsummer**
This adds the concept of SimpleTables.get_stat_data and uses it for the ECD data (and for elections, but that's not currently shown).

So this is the proof of concept that get_stat_data is useful and simplifies things.

@xybrnet

### CODE REVIEW

File: api/models/tables.py [hunk 1]
```diff
@@ -139,6 +145,92 @@ def raw_data_for_geos(self, geos):
...
+
+            recode = recode or {}
+            if recode:
+                # change lambda to dicts
+                if not isinstance(recode, dict):
+                    recode = {f: recode(f) for f in fields}
+
+            # table columns to fetch
+            cols = [self.model.columns[c] for c in fields]
+            if total is not None and isinstance(total, basestring) and total not in cols:
```
Comment (xybrnet) [inline api/models/tables.py]: Should we not also check whether total is in self.columns?

File: api/models/tables.py [hunk 2]
```diff
@@ -139,6 +145,92 @@ def raw_data_for_geos(self, geos):
...
+
+                if percent:
+                    # sum up existing values, if any
+                    val = val + results.get('numerators', {}).get('this', 0)
+                    results[key]['values'] = {'this': p(val, total)}
+                    results[key]['numerators'] = {'this': val}
+                else:
+                    # sum up existing values, if any
+                    val = val + results.get('values', {}).get('this', 0)
+                    results[key]['values'] = {'this': val}
```
Comment (xybrnet) [inline api/models/tables.py]: I'm not sure I understand what is happening here. Could we maybe have a quick chat about this a bit later?

File: api/models/tables.py (hunk 1 above)
Comment (longhotsummer) [inline api/models/tables.py]: Good point, pushing a fix.

File: api/models/tables.py (hunk 2 above)
Comment (longhotsummer) [inline api/models/tables.py]: Sure. I've also added some extra comments to try help make it clearer.

**Pull request merged**